modules/apps/depot/depot-web/src/main/java/com/liferay/depot/web/internal/util/DepotEntryAdminSearchProvider.java

```diff
@@ -0,0 +1,192 @@
+				depotGroupItemSelectorCriterion.getDepotEntryType(),
+				portletRequest, portletResponse, portletURL);
+		}
+
+		return _getDepotEntrySearch(
+			depotGroupItemSelectorCriterion.getDepotEntryType(), portletRequest,
+			portletResponse, portletURL);
+	}
+
+	public DepotEntrySearch getDepotEntrySearch(
+			int depotEntryType, PortletRequest portletRequest,
+			PortletResponse portletResponse, PortletURL portletURL)
+		throws PortalException {
+
+		return _getDepotEntrySearch(
+			depotEntryType, portletRequest, portletResponse, portletURL);
+	}
+
+	private DepotEntrySearch _getDepotEntrySearch(
+			int depotEntryType, PortletRequest portletRequest,
+			PortletResponse portletResponse, PortletURL portletURL)
+		throws PortalException {
+
+		DepotEntrySearch depotEntrySearch = new DepotEntrySearch(
+			portletRequest, portletResponse, portletURL, "depotEntries");
+
+		depotEntrySearch.setEmptyResultsMessage(
+			_language.get(
+				portletRequest.getLocale(), "no-asset-libraries-were-found"));
+
+		depotEntrySearch.setResultsAndTotal(
+			() -> _getResults(
+				depotEntryType, portletRequest, depotEntrySearch.getStart(),
+				depotEntrySearch.getEnd()),
+			_getTotal(depotEntryType, portletRequest));
+
+		return depotEntrySearch;
+	}
+
+	private DepotEntrySearch _getGroupConnectedDepotEntrySearch(
+			int depotEntryType, PortletRequest portletRequest,
+			PortletResponse portletResponse, PortletURL portletURL)
+		throws PortalException {
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)portletRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		DepotEntrySearch depotEntrySearch = new DepotEntrySearch(
+			portletRequest, portletResponse, portletURL, "depotEntries");
+
+		depotEntrySearch.setEmptyResultsMessage(
+			_language.get(
+				portletRequest.getLocale(),
+				(depotEntryType == DepotConstants.TYPE_SPACE) ?
+					"no-spaces-were-found" : "no-asset-libraries-were-found"));
+
+		depotEntrySearch.setResultsAndTotal(
+			() -> _depotEntryService.getGroupConnectedDepotEntries(
+				themeDisplay.getScopeGroupId(), depotEntryType,
+				depotEntrySearch.getStart(), depotEntrySearch.getEnd()),
+			_depotEntryService.getGroupConnectedDepotEntriesCount(
+				themeDisplay.getScopeGroupId(), depotEntryType));
+
+		return depotEntrySearch;
+	}
+
+	private List<DepotEntry> _getResults(
+			int depotEntryType, PortletRequest portletRequest, int start,
+			int end)
+		throws PortalException {
+
+		List<DepotEntry> depotEntries = new ArrayList<>();
+
+		Indexer<Object> indexer = IndexerRegistryUtil.getIndexer(
+			DepotEntry.class.getName());
+
+		Hits hits = indexer.search(
+			_getSearchContext(depotEntryType, portletRequest, start, end));
+
+		for (Document document : hits.getDocs()) {
+			long classPK = GetterUtil.getLong(
+				document.get(Field.ENTRY_CLASS_PK));
+
+			DepotEntry depotEntry = _depotEntryService.getDepotEntry(classPK);
+
+			if (depotEntry != null) {
+				depotEntries.add(depotEntry);
+			}
```

Comment:
You're invoking a `get*` method, so the nullness check is redundant (it will either return a nonnull value or fail with an exception).

Reply:
true, is the rest good to go?